### PR TITLE
feat(account): Add unique constraint for a user's primary email

### LINKED ISSUE #225: Data model allows for having multiple primary e-mail addresses
As discussed in #210, having at most one primary e-mail address is currently not enforced at the database level. To be determined: should we fix this, e.g. by making user and primary "unique together" and using primary=NULL instead of False.

### PR DESCRIPTION
**Pull request by jnns**
Since Django 2.2 the UniqueConstraint supports conditions. This allows enforcing only one primary email per user on the database level.

The docs mention that [Oracle databases don't support partial indexes](https://docs.djangoproject.com/en/2.2/ref/models/indexes/#django.db.models.Index.condition). I'm not sure if that means that the index will silently be ignored (just as MariaDB and MySQL do).

Closes #225.

# Submitting Pull Requests

## General

 - [x] Make sure you use [semantic commit messages](https://seesparkbox.com/foundry/semantic_commit_messages).
       Examples: `"fix(google): Fixed foobar bug"`, `"feat(accounts): Added foobar feature"`.
 - [x] All Python code must formatted using Black, and clean from pep8 and isort issues.
 - [ ] JavaScript code should adhere to [StandardJS](https://standardjs.com).
 - [x] If your changes are significant, please update `ChangeLog.rst`.
 - [ ] If your change is substantial, feel free to add yourself to `AUTHORS`.

### CODE REVIEW

File: allauth/account/adapter.py [hunk 1]
```diff
@@ -483,15 +484,11 @@ def set_password(self, user, password):
...
-                app_settings.USER_MODEL_USERNAME_FIELD,
-                "first_name",
-                "last_name",
-                "email",
-            ],
-        )
+        return [
+            f.name
+            for f in user._meta.fields
+            if type(f) in [CharField, EmailField] and f.name != "password"
```
Comment (pennersr) [inline allauth/account/adapter.py:490]: The changes made here don't seem to be related to the PR title:

>  feat(account): Add unique constraint for a user's primary email #2996

Comment (derek-adair) [reply allauth/account/adapter.py:490]: Pretty nonsensical PR - close this.  Looks like they crossed feature branches or something...
